### ISSUE TITLE
[Android] IllegalArgumentException on Sentry initialization

### ISSUE THREAD

**Issue by TimBurik** [getsentry/sentry-dotnet#3853]
### Package

Sentry

### .NET Flavor

.NET

### .NET Version

9.0.100

### OS

Android

### SDK Version

4.13.0

### Steps to Reproduce

1. Initialize Sentry 4.13.0 using a valid DSN

### Expected Result

Sentry is initialized, up and running

### Actual Result

Exception is thrown:
```
Java.Lang.IllegalArgumentException: DSN is required. Use empty string or set enabled to false in SentryOptions to disable SDK.
   at Java.Interop.JniEnvironment.StaticMethods.CallStaticVoidMethod(JniObjectReference type, JniMethodInfo method, JniArgumentValue* args)
   at Java.Interop.JniPeerMembers.JniStaticMethods.InvokeVoidMethod(String encodedMember, JniArgumentValue* parameters)
   at Sentry.JavaSdk.Android.Core.SentryAndroid.Init(Context context, IOptionsConfiguration configuration)
   at Sentry.SentrySdk.InitSentryAndroidSdk(SentryOptions options)
   at Sentry.SentrySdk.InitHub(SentryOptions options)
   at Sentry.SentrySdk.Init(SentryOptions options)
   at Sentry.SentrySdk.Init(Action`1 configureOptions)
   at SentryPlayground.Issues.InitializationException.Android.MainApplication.OnCreate()
   at Android.App.Application.n_OnCreate(IntPtr jnienv, IntPtr native__this)
   at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz)
  --- End of managed Java.Lang.IllegalArgumentException stack trace ---
java.lang.IllegalArgumentException: DSN is required. Use empty string or set enabled to false in SentryOptions to disable SDK.
	at io.sentry.Sentry.initConfigurations(Sentry.java:396)
	at io.sentry.Sentry.init(Sentry.java:234)
	at io.sentry.Sentry.init(Sentry.java:167)
	at io.sentry.android.core.SentryAndroid.init(SentryAndroid.java:93)
	at io.sentry.android.core.SentryAndroid.init(SentryAndroid.java:76)
	at crc64a0ace0f45caee1c5.MainApplication.n_onCreate(Native Method)
	at crc64a0ace0f45caee1c5.MainApplication.onCreate(MainApplication.java:25)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1316)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7685)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2478)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:230)
	at android.os.Looper.loop(Looper.java:319)
	at android.app.ActivityThread.main(ActivityThread.java:8893)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)

  --- End of managed Java.Lang.IllegalArgumentException stack trace ---
java.lang.IllegalArgumentException: DSN is required. Use empty string or set enabled to false in SentryOptions to disable SDK.
	at io.sentry.Sentry.initConfigurations(Sentry.java:396)
	at io.sentry.Sentry.init(Sentry.java:234)
	at io.sentry.Sentry.init(Sentry.java:167)
	at io.sentry.android.core.SentryAndroid.init(SentryAndroid.java:93)
	at io.sentry.android.core.SentryAndroid.init(SentryAndroid.java:76)
	at crc64a0ace0f45caee1c5.MainApplication.n_onCreate(Native Method)
	at crc64a0ace0f45caee1c5.MainApplication.onCreate(MainApplication.java:25)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1316)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7685)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2478)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:230)
	at android.os.Looper.loop(Looper.java:319)
	at android.app.ActivityThread.main(ActivityThread.java:8893)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
```

Interestingly, this issue happens not in every Android project, which initializes Sentry 4.13.0. Just in case, I'm attaching a sample where I have 100% reproduction rate. Also, this issue is not reproduced with Sentry 4.12.2
[SentryPlayground.Issues.InitializationException.Android.zip](https://github.com/user-attachments/files/18179970/SentryPlayground.Issues.InitializationException.Android.zip)

It seems that some kind of exception is thrown while converting SentryOptions to Java, so the DSN value is not actually applied. 
https://github.com/getsentry/sentry-java/blob/90eb1e3c65f721893b17bce4d286b482afccacb8/sentry/src/main/java/io/sentry/Sentry.java#L193-L202
Although I can't find an error message in logs (maybe logger is not initialized at that point)

**Comment by jamescrosswell** [comment]
Hi @TimBurik ,

I think you might be seeing this error:
- https://github.com/getsentry/sentry-dotnet/issues/3755

This was introduced when Session Replay was added in version `4.13.0`. As a temporary workaround, you could revert to version `4.12.2`.

**Comment by TimBurik** [comment]
Thank you, @jamescrosswell 
Indeed, this issue seems to be a duplicate of the https://github.com/getsentry/sentry-dotnet/issues/3755, sorry about that. Hopefully, it contains some useful additional information though.

I would like to highlight, that we have one Android project which is currently uses Sentry 4.13.0 and is not affected by this issue (although we probably should roll back to 4.12.2 just to be on the safe side, since we don't use replay functionality). Maybe it relates to the additional Sentry settings we use, or even to project settings - I'll try to reproduce this behavior on sample project

**Comment by TimBurik** [comment]
@jamescrosswell done! I managed to make the sample work by simply adding package dependency (have not tested other versions of the package though):
`<PackageReference Include="Xamarin.AndroidX.AppCompat" Version="1.6.1.7" />`

[SentryPlayground.Issues.InitializationException.Android_workaround.zip](https://github.com/user-attachments/files/18197693/SentryPlayground.Issues.InitializationException.Android_workaround.zip)

**Comment by jamescrosswell** [comment]
@TimBurik you're a rock star! It works... I have no idea how or why yet, but I think this should be enough to unblock https://github.com/getsentry/sentry-dotnet/issues/3755.

Thank you so much 🙏🏻

**Comment by jamescrosswell** [comment]
Closing this issue in favour of #3755, which contains a detailed explanation of the problem.

Thanks again for your help with this @TimBurik !
